### PR TITLE
fix test

### PR DESCRIPTION
**Pull request by Un1q32**
Mistake with #78628 that got caught after being merged

### CODE REVIEW

File: lld/test/MachO/link-csu-object.s [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # RUN: mkdir -p %t
 # RUN: llvm-mc -filetype=obj -triple=x86_64-apple-darwin %p/Inputs/libhello.s -o %t/hello.o
 # RUN: llvm-mc -filetype=obj -triple=x86_64-apple-darwin %s -o %t/main.o
-# RUN: %lld -L %t %t/main.o %t/hello.o -o %t/a.out
+# RUN: %lld -L %t %t/main.o -lhello.o -o %t/a.out
```
Comment (oontvoo) [inline lld/test/MachO/link-csu-object.s:5]: I think @smeenai was suggesting  sth like:

`%lld -lhello %t/main.o -o %t/a.out` (ie., using `-l<lib_name>` syntax - __without__ the `.o`). 

Perhaps for easy comparison, I'd suggest leaving your line #5  alone - and add an additional link that use the `-l<lib_name>` syntax - then smoke-check that the two output are the same.
(@smeenai thoughts?)

Comment (OldWorldOrdr) [reply lld/test/MachO/link-csu-object.s:5]: The point of the pr was for Csu support files, which specify the whole name of the object, `-lhello` wont link hello.o

This fixes building with MacOS X 10.7 and 10.6 SDKs which require linking with files like crt1.o, clang passes `-lcrt1.o` to the linker, which works with ld64 but not lld

Comment (oontvoo) [reply lld/test/MachO/link-csu-object.s:5]: D'oh . Thanks !

**Comment by oontvoo** [comment]
(Will merge after the pending checks finish)

**Review by smeenai — APPROVED** [review]
Thank you!

**Pull request merged**